src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferService.java

```diff
@@ -1,47 +1,210 @@
-   * @param HIPHealthInformationRequest HIU public keys and dataPush URL is provided
+   *
+   * @param hipHealthInformationRequest HIU public keys and dataPush URL is provided
    */
   @Override
-  public void requestOnReceived(HIPHealthInformationRequest HIPHealthInformationRequest) {
-    // TODO Implement onRequest workflow logic.
-    initiateBundleRequest(HIPHealthInformationRequest);
+  public void
+  requestOnReceived(HIPHealthInformationRequest hipHealthInformationRequest) {
+    HIPHealthInformationRequestAcknowledgement hipHealthInformationRequestAcknowledgement = null;
+    RespRequest responseRequestId =
+        RespRequest.builder().requestId(hipHealthInformationRequest.getRequestId()).build();
+    String consentId = hipHealthInformationRequest.getHiRequest().getConsent().getId();
+    HiRequest hiRequest =
+        HiRequest.builder()
+            .sessionStatus("ACKNOWLEDGED")
+            .transactionId(hipHealthInformationRequest.getTransactionId())
+            .build();
+    if (logsRepo.findByConsentId(consentId) != null) {
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .hiRequest(hiRequest)
+              .resp(responseRequestId)
+              .build();
+    } else {
+      errorResponse.setMessage("HIP -> ConsentId not found in cache");
+      errorResponse.setCode(1000);
+      log.error("DataOnRequest body -> making error body, consentId not found in cache");
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .error(errorResponse)
+              .resp(responseRequestId)
+              .build();
+    }
+    log.info(hipHealthInformationRequestAcknowledgement.toString());
+    try {
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(
+              dataOnRequestPath, hipHealthInformationRequestAcknowledgement);
+      log.debug(dataOnRequestPath + " : dataOnRequest: " + response.getStatusCode());
+      if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_SUCCESS,
+            hipHealthInformationRequestAcknowledgement);
+      } else if (Objects.nonNull(response.getBody())
+          && Objects.nonNull(response.getBody().getErrorResponse())) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_ERROR,
+            hipHealthInformationRequestAcknowledgement);
+      }
+    } catch (Exception ex) {
+      String error =
+          "Exception while Initiating HIP auth: "
```

Comment:
Change the error message to `An unknown error occurred while calling Gateway API`